Library/PhotosController.swift

```diff
@@ -84,17 +98,30 @@ extension PhotosController {
         guard let collectionView = self.collectionView else { return }
 
         self.viewerController = ViewerController(initialIndexPath: indexPath, collectionView: collectionView)
+        self.viewerController!.dataSource = self
+
         #if os(iOS)
             let headerView = HeaderView()
             headerView.viewDelegate = self
             self.viewerController?.headerView = headerView
             let footerView = FooterView()
             footerView.viewDelegate = self
             self.viewerController?.footerView = footerView
+        #else
+            self.viewerController!.delegate = self
         #endif
-        self.viewerController!.dataSource = self
+
         self.present(self.viewerController!, animated: false, completion: nil)
     }
+
+    #if os(tvOS)
+    override public func collectionView(_ collectionView: UICollectionView, canFocusItemAt indexPath: IndexPath) -> Bool {
```

Comment:
If the viewer is dismissed it will work as usual, otherwise it ignores the focus changes, this disables the default behavior where the collection view will try to figure out which view to focus on, and instead we will tell the collection view which view to focus thanks to the `preferredFocusEnvironments`.